src/core/bitop.d

```diff
@@ -706,3 +706,38 @@ version (D_InlineAsm_X86_64)
         }
     }
 }
+
+/**
+ *  Bitwise rotates unsigned arithmetic types
+ */
+pure T rotl(T)(in T u, in uint s)
+    if (__traits(isArithmetic, T) && __traits(isUnsigned, T))
+{
+    return cast(T) ((u << s) | (u >> (-s & (T.sizeof * 8 - 1))));
+}
+
+pure T rotr(T)(in T u, in uint s)
+    if (__traits(isArithmetic, T) && __traits(isUnsigned, T))
+{
+    return cast(T) ((u >> s) | (u << (-s & (T.sizeof * 8 - 1))));
+}
+
+unittest
```

Comment:
Make this a [documented unittest](http://dlang.org/spec/unittest.html#documented-unittests).